pkg/tfbridge/adapt_check_failures.go

```diff
@@ -67,20 +67,13 @@ func parseCheckError(
 	schemaInfos map[string]*SchemaInfo,
 	err error,
 ) (*CheckFailurePath, CheckFailureReason, string) {
-	if parts := requiredFieldRegex.FindStringSubmatch(err.Error()); len(parts) == 2 {
```

Comment:
Looks like this was trying to regex-match err.Error() on a ValidationError but we can do better. In particular by type-matching we can avoid having to guess the Path to a property.